dev/release/README.md

```diff
@@ -0,0 +1,106 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Release
+
+## Overview
+
+    1. Test the revision to be released
+    2. Prepare RC and vote (detailed later)
+    3. Publish (detailed later)
+
+### Prepare RC and vote
+
+Run `dev/release/release_rc.sh` on a working copy of 
+`git@github.com:apache/iceberg-go` not from your fork:
+
+```console
+$ git clone git@github.com:apache/iceberg-go.git
+$ dev/release/release_rc.sh ${RC}
```

Comment:
should this maybe also take a version number?

Reply:
yea, you're right. As a TODO maybe we can use `go-semantic-release` to have it automatically figure out the next version. But that's something that we can leave for a future update to the scripts. For now it can take the version number as an argument.